skbio/sequence/_sequence.py

```diff
@@ -187,30 +351,107 @@ def metadata(self):
     def metadata(self, metadata):
         if not isinstance(metadata, dict):
             raise TypeError("metadata must be a dict")
-        self._metadata = metadata
+        # shallow copy
+        self._metadata = metadata.copy()
+
+    @metadata.deleter
+    def metadata(self):
+        if hasattr(self, '_metadata'):
+            del self._metadata
+        self.metadata = {}
 
     @property
     def positional_metadata(self):
-        """Pandas DataFrame containing metadata on a per-character basis.
+        """pandas.DataFrame containing metadata on a per-character basis.
 
         Notes
         -----
-        This property is writeable.
+        This property can be set and deleted.
 
         Examples
         --------
+        Create a sequence with positional metadata:
+
         >>> from skbio import Sequence
-        >>> s = Sequence('ACGTACGTACGTACGT',
-        ...              positional_metadata={'quality': range(16)})
+        >>> s = Sequence(
+        ...     'ACGT',
+        ...     positional_metadata={'quality': [3, 3, 20, 11],
+        ...                          'exons': [True, True, False, True]})
         >>> s # doctest: +NORMALIZE_WHITESPACE
-        Sequence('ACGTACGTACGTACGT', length=16, has_metadata=False,
+        Sequence('ACGT', length=4, has_metadata=False,
                  has_positional_metadata=True)
-        >>> s.positional_metadata['quality'].values
-        array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15])
+
+        Retrieve positional metadata:
+
+        >>> s.positional_metadata
+           exons  quality
+        0   True        3
+        1   True        3
+        2  False       20
+        3   True       11
+
+        Update positional metadata:
+
+        >>> s.positional_metadata['gaps'] = [False, False, False, False]
```

Comment:
Would it be worth using `DNA` for this example so that you could use `s.positional_metadata['gaps'] = s.gaps()`?